unstract/connectors/src/unstract/connectors/filesystems/unstract_file_system.py

```
@@ -103,6 +103,116 @@ def is_dir_by_metadata(self, metadata: dict[str, Any]) -> bool:
+
+        for root, dirs, _ in fs_fsspec.walk(directory, maxdepth=max_depth):
+            try:
+                fs_metadata_list = fs_fsspec.listdir(root)
+                for metadata in fs_metadata_list:
+                    if not include_dirs and self.is_dir_by_metadata(metadata):
+                        continue
+                    # Add context for later processing if required
+                    metadata["_root_dir"] = root
+                    metadata["_dirs"] = dirs
+                    all_files.append(metadata)
+            except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)